### PR TITLE
Adds `TRANSFORMERS_TEST_DEVICE`

### PR DESCRIPTION
**Pull request by vvvm23**
Adds support for environment variable `TRANSFORMERS_TEST_DEVICE` to set device in use for running the test suite. This pattern is already in use in `diffusers`.

# What does this PR do?

Adds support for environment variable `TRANSFORMERS_TEST_DEVICE` to set device in use for running the test suite. This is a pattern already in use in [`diffusers`](https://github.com/huggingface/diffusers/blob/d93ca268938940839118fdb8d0cf4c1ca7a9fead/src/diffusers/utils/testing_utils.py#L45) which would be useful to have in transformers.

Additionally, I would like to propose removing the check on available backends, as is found in the diffusers version. I included it here to match diffusers, but it would be useful to remove it (for example, if testing new backends and the like). Let me know if that is okay and I will amend the PR.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [X] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

@sgugger, git blame says you! :hugs:

### CODE REVIEW

**Review by sgugger — COMMENTED** [review]
Thanks for your PR! Instead of doing a test on backends which may or may not be available, it would be better to just try to do `torch.device` with the device found, to see if it errors or not.

**Comment by vvvm23** [comment]
Hi thanks, it is a good suggestion to simply try creating the device. Perhaps `diffusers` can be updated to do the same?

Doing this approach makes the list of backends obsolete, so I removed. I just let it throw an unhandled error as I feel the message is informative enough, but let me know if you want that to be caught.

File: src/transformers/testing_utils.py [hunk 1]
```diff
@@ -614,7 +614,11 @@ def require_torch_multi_npu(test_case):
     # Set env var CUDA_VISIBLE_DEVICES="" to force cpu-mode
     import torch
 
-    if torch.cuda.is_available():
+    if "TRANSFORMERS_TEST_DEVICE" in os.environ:
+        torch_device = os.environ["TRANSFORMERS_TEST_DEVICE"]
+        # try creating device to see if provided device is valid
+        _ = torch.device(torch_device)
```
Comment (sgugger) [inline src/transformers/testing_utils.py:620]: Maybe a nice try except here to throw a helpful error message? But yes better this way!

**Review by sgugger — COMMENTED** [review]
Thanks for iterating! One last nit and we should be good to merge!

**Comment by vvvm23** [comment]
Updated the error message, let me know if it works for you~

**Review by sgugger — APPROVED** [review]
Thanks a lot! Can you just run `make style` on your branch to fix the quality issue?

**Review by ArthurZucker — COMMENTED** [review]
Could you also add some documentation in `testing.md` to not make it hidden?

**Comment by vvvm23** [comment]
> Thanks a lot! Can you just run make style on your branch to fix the quality issue?

Woops, I ran this but forgot to commit the changes.

> Could you also add some documentation in testing.md to not make it hidden?
Will do 👍 where do you think the best place is this for this in the file?

I should say, to all the suggested changes here, should these also be mirrored in `diffusers`? My initial PR was basically a direct copy from there.

**Comment by ArthurZucker** [comment]
You can probably add it to `docs/source/en/testing.md`.
Not sure about diffusers but documenting a new env variable should always be good 😉

**Comment by vvvm23** [comment]
@ArthurZucker added a small section following the "To GPU or not to GPU" section – in my head this fits well. Let me know how you feel about the language there 🤗 I ran `make style` also which I believe also formats the docs.

**Review by ArthurZucker — APPROVED** [review]
Thanks! Just waiting for the CIs

**Pull request merged**